plugin/utils/gapic_utils.py

```diff
@@ -251,19 +261,14 @@ def update_collections(
 
         parent_collection_names = []
         for pattern in parent_patterns:
-            collections.setdefault(name, {}).update({
-                'entity_name': parent_entity_names[pattern],
-                'name_pattern': pattern,
+            collections.update({
+                parent_entity_names[pattern]: {
+                    'entity_name': parent_entity_names[pattern],
+                    'name_pattern': pattern,
+                }
             })
             parent_collection_names.append(parent_entity_names[pattern])
 
-        if has_oneof:
-            # Add the resource collection to "collection_oneofs".
-            collection_oneofs.setdefault(name, {}).update({
-                'oneof_name': 'parent_oneof',
```

Comment:
I'm not sure in what case we would expect the oneof name to be exactly `parent_oneof`